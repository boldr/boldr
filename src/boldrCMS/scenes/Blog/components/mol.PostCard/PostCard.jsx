import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import Moment from 'moment';
import TagIcon from 'material-ui/svg-icons/action/bookmark';
import RaisedButton from 'material-ui/RaisedButton';
import { Card, CardActions, CardHeader, CardMedia, CardTitle, CardText } from 'material-ui/Card';
import classNames from 'classnames/bind';
import styles from './style.css';

const cx = styles::classNames;

const PostCard = props => {
  const formattedDate = Moment(props.createdAt).format('MMMM Do YYYY, h:mm:ss a');
  const profileLink = `profile/public/${props.user.id}`;
  return (
    <div>
      <Card>
        <CardMedia overlay={
          <CardTitle title={ props.title } subtitle={ formattedDate } />
        }>
          <img className={ cx('post__card-image') } src={ props.featureImage } height="350px" />
        </CardMedia>
        <CardText>
        { props.excerpt }
        </CardText>
        <div className={ cx('post__card-footer') }>
          <div className="grid__row">
            <div className="grid__twothirds">
              <ul style={ { display: 'flex', listStyleType: 'none' } }>
                <li>
                  <TagIcon />
                </li>
                 { props.tags.map(tag =>
                    <li key={ tag.id }>
                      <Link to={ `/blog/tags/${tag.id}` } >{ tag.name } &nbsp;</Link>
                    </li>)
                 }
              </ul>
            </div>
            <div className="grid__third">
              <Link to={ `/blog/${props.slug}` }>
                <RaisedButton label="Continue" secondary />
              </Link>
            </div>
          </div>
          Posted by <Link to={ profileLink }>{ props.user.displayName }</Link>
        </div>
      </Card>
      </div>
    );
};

PostCard.propTypes = {
  title: PropTypes.string,
  createdAt: PropTypes.string,
  displayName: PropTypes.string,
  featureImage: PropTypes.string,
  excerpt: PropTypes.string,
  tags: PropTypes.array,
  content: PropTypes.string,
  user: PropTypes.object,
  slug: PropTypes.string
};

export default PostCard;
