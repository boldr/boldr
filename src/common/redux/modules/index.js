import { combineReducers } from 'redux';
import { routerReducer as routing } from 'react-router-redux';
import auth from './auth/auth.reducer';

const rootReducer = combineReducers({
  routing,
  auth
});

export default rootReducer;
