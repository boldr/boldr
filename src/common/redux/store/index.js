let store;

if (__DEV__) {
  store = require('./configureStore.dev').default;
} else {
  store = require('./configureStore.prod').default;
}

export default store;
