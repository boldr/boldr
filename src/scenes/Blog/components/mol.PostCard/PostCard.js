import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import dateFns from 'date-fns';
import classNames from 'classnames/bind';
import { Button, Card, Image } from 'stardust';
import TagBlock from '../mol.TagBlock';
import FontIcon from 'components/md/FontIcons';
import { FlatButton, IconButton } from '../../../../components/md/Buttons';
// import { Card, CardMedia, CardTitle, CardActions, CardText } from '../../../../components/md/Cards';
import Avatar from '../../../../components/md/Avatars';
import { Row } from '../../../../components';

const cx = styles::classNames;

const PostCard = props => {
  const formattedDate = dateFns.format(props.created_at, 'MM/DD/YYYY');
  // const profileLink = `profile/public/${props.author.id}`;
  // const overlay = (
  //   <CardTitle
  //     key="overlay"
  //     title={ props.title }
  //     subtitle={ formattedDate }
  //   >        <CardTitle
  //             avatar={ <Link to={`/profile/public/${props.author.id}`}><Avatar src={ props.author.avatar_url } alt="Author avatar image" /></Link> }
  //             title={ props.author.display_name }
  //           />
  //     <IconButton className="margin-left-auto">star_outline</IconButton>
  //   </CardTitle>
  // );
  return (
    <div>
      <Card fluid>

          <Image className={ cx('post__card-image') } src={ props.feature_image } />
          <Card.Content>
          <Card.Header>
          { props.title }

        </Card.Header>
          <Card.Meta>
            { formattedDate }
          </Card.Meta>
       <Card.Description>
        { props.excerpt }
         </Card.Description>
        </Card.Content>
        <Card.Content extra>
        <Row>
        <Link to={ `/blog/${props.slug}` }>
          <FlatButton label="Read more" secondary />
        </Link>
        <TagBlock tags={ props.tags } />
        </Row>
        </Card.Content>
      </Card>
      </div>
    );
};

PostCard.propTypes = {
  title: PropTypes.string,
  created_at: PropTypes.string,
  display_name: PropTypes.string,
  feature_image: PropTypes.string,
  excerpt: PropTypes.string,
  tags: PropTypes.array,
  content: PropTypes.string,
  author: PropTypes.object,
  slug: PropTypes.string
};

export default PostCard;
