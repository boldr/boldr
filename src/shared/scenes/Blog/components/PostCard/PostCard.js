/* @flow */
import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import dateFns from 'date-fns';
import classnames from 'classnames';
import Button from 'react-md/lib/Buttons/Button';
import Card from 'react-md/lib/Cards/Card';
import CardTitle from 'react-md/lib/Cards/CardTitle';
import CardActions from 'react-md/lib/Cards/CardActions';
import CardText from 'react-md/lib/Cards/CardText';
import { Col, Row, StyleClasses, Media, MediaOverlay } from 'boldr-ui';
import { selectPost } from '../../../../state/modules/blog/posts/actions';

import TagBlock from '../TagBlock';

const BASE_ELEMENT = StyleClasses.POST_CARD;
type Props = {
  className: string,
  id: ?string,
  featureImage: ?string,
  title: string,
  slug: string,
  content: ?string,
  backgroundImage: ?string,
  excerpt: ?string,
  createdAt: string,
  updatedAt: ?string,
  status: ?string,
  author: string,
  comments: ?Array<Object>,
  seo: ?Object,
  tags: ?Array<Tag>,
  attachments: ?Object,
  meta: ?Object,
  userId: ?string,
  dispatch: Function,
  listTags: Object,
};

export const PostCard = (props: Props) => {
  const formattedDate = dateFns.format(props.createdAt, 'MM/DD/YYYY');
  const classes = classnames(BASE_ELEMENT, props.className);
  const postTags = props.tags ? props.tags.map(id => props.listTags[id]) : null;
  // Explicitly define post rather than passing additional
  // unnecessary props like listTags
  const post = {
    id: props.id,
    author: props.author,
    attachments: props.attachments,
    content: props.content,
    createdAt: props.createdAt,
    excerpt: props.excerpt,
    backgroundImage: props.backgroundImage,
    featureImage: props.featureImage,
    comments: props.comments,
    meta: props.meta,
    slug: props.slug,
    status: props.status,
    tags: props.tags,
    title: props.title,
    userId: props.userId,
  };
  function transitionPost() {
    props.dispatch(selectPost(post));
  }

  return (
    <div className={ classes }>
      <Card>
        <Media>
          <img src={ props.featureImage } alt={ `${props.title} feature` } role="presentation" />
          <MediaOverlay>
            <CardTitle title={ props.title } subtitle={ formattedDate }>
              <Button className="md-cell--right" icon>star_outline</Button>
            </CardTitle>
          </MediaOverlay>
        </Media>
        <CardText>
          {props.excerpt}
          <Row>
            <Col xs={ 12 }>
              <Link to={ `/blog/${props.slug}` } className="readmore-link">
                <Button raised primary label="Read More" onClick={ transitionPost } />
              </Link>
            </Col>
          </Row>
        </CardText>
        <CardActions>
          <TagBlock tags={ postTags } />
        </CardActions>
      </Card>
    </div>
  );
};

export default connect()(PostCard);
