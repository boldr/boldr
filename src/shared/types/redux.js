/* @flow */
import type { Store as ReduxStore } from 'redux';

export type ApiAction<T> = {
  key: string,
  payload: T,
};

export type ActionType =
  | 'LOGIN_REQUEST'
  | 'LOGIN_SUCCESS'
  | 'LOGIN_FAILURE'
  | 'LOGOUT_USER'
  | 'LOGOUT_USER_FAIL'
  | 'CHECK_AUTH_REQUEST'
  | 'CHECK_AUTH_SUCCESS'
  | 'CHECK_AUTH_FAILURE'
  | 'SIGNUP_USER_REQUEST'
  | 'SIGNUP_USER_SUCCESS'
  | 'SIGNUP_USER_FAILURE'
  | 'FORGOT_PASSWORD_REQUEST'
  | 'FORGOT_PASSWORD_SUCCESS'
  | 'FORGOT_PASSWORD_FAILURE'
  | 'RESET_PASSWORD_REQUEST'
  | 'RESET_PASSWORD_SUCCESS'
  | 'RESET_PASSWORD_FAILURE'
  | 'LOAD_POSTS_REQUEST'
  | 'LOAD_POSTS_SUCCESS'
  | 'LOAD_POSTS_FAILURE'
  | 'FETCH_POST_REQUEST'
  | 'FETCH_POST_SUCCESS'
  | 'FETCH_POST_FAILURE'
  | 'UPDATE_POST_REQUEST'
  | 'UPDATE_POST_SUCCESS'
  | 'UPDATE_POST_FAILURE'
  | 'SELECT_POST'
  | 'SELECT_POST_SUCCESS'
  | 'SELECT_POST_FAILURE'
  | 'CREATE_POST_REQUEST'
  | 'CREATE_POST_SUCCESS'
  | 'CREATE_POST_FAILURE'
  | 'DELETE_POST_FAILURE'
  | 'DELETE_POST_REQUEST'
  | 'DELETE_POST_SUCCESS';

export type PostsReducer = {
  all: Object,
  ids: Array<Object>,
  isFetching: boolean,
  currentPost: Object,
};

export type BlogReducer = {
  posts: PostsReducer,
  tags: any,
  comments: any,
};

export type UserInfo = {
  [userId: string]: {
    readyStatus: string,
    err: any,
    info: Object,
  },
};

export type Reducer = {
  routing: any,
  blog: BlogReducer,
};


export type Action =
  { type: '@boldr/blog/FETCH_POSTS_REQUEST' } |
  { type: '@boldr/blog/FETCH_POSTS_SUCCESS', data: Array<Object> } |
  { type: '@boldr/blog/FETCH_POSTS_FAILURE', err: any } |
  { type: '@boldr/blog/FETCH_POST_REQUEST', userId: string } |
  { type: '@boldr/blog/FETCH_POST_SUCCESS', userId: string, data: Object } |
  { type: '@boldr/blog/FETCH_POST_FAILURE', userId: string, err: any };

export type Store = ReduxStore<Reducer, Action>;
// eslint-disable-next-line no-use-before-define
export type Dispatch = (action: Action | ThunkAction | PromiseAction | Array<Action>) => any;
export type GetState = () => Object;
export type ThunkAction = (dispatch: Dispatch, getState: GetState) => any;
export type PromiseAction = Promise<Action>;
