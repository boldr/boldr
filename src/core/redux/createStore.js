import { createStore as _createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import thunkMiddleware from 'redux-thunk';
import createLogger from 'redux-logger';
import rootReducer from '../../state';
import createMiddleware from './clientMiddleware';

const ISDEV = process.env.NODE_ENV === 'development';
export default function createStore(history, client, data) {
  // Sync dispatched route actions to the history
  const reduxRouterMiddleware = routerMiddleware(history);
  const logger = createLogger();
  const middleware = [thunkMiddleware, createMiddleware(client), reduxRouterMiddleware, logger];

  let finalCreateStore;
  if (ISDEV) {
    finalCreateStore = compose(
      applyMiddleware(...middleware),
      typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f // eslint-disable-line
    )(_createStore);
  } else {
    finalCreateStore = applyMiddleware(...middleware)(_createStore);
  }

  const store = finalCreateStore(rootReducer, data);


  if (ISDEV && module.hot) {
    module.hot.accept('../../state', () => {
      const nextReducer = require('../../state');

      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
