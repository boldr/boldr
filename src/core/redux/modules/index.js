import { combineReducers } from 'redux';
import { routerReducer as routing } from 'react-router-redux';
import auth from './auth/auth.reducer';

const rootReducer = combineReducers({
  auth,
  routing
});

export default rootReducer;
