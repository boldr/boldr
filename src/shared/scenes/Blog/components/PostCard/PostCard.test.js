import React from 'react';
import {createStore} from 'redux';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {PostCard} from './PostCard';

describe('<PostCard />', () => {
  it('renders <PostCard /> without breaking', () => {
    const wrapper = shallow(<PostCard title="blah" />);
    expect(wrapper.find('div').length).toBe(1);
  });

  it('accepts props and renders them.', () => {
    const wrapper = shallow(
      <PostCard
        feature_image="http://boldr.io/images/logo.png"
        title="Test Post"
        excerpt="An excerpt"
        slug="test-post"
      />,
    );
    expect(wrapper.instance().props.feature_image).toBe(
      'http://boldr.io/images/logo.png',
    );
    expect(wrapper.instance().props.title).toBe('Test Post');
    expect(wrapper.instance().props.excerpt).toBe('An excerpt');
    expect(wrapper.instance().props.slug).toBe('test-post');
  });
  it('renders snapshot', () => {
    const wrapper = shallow(<PostCard title="Test Post" />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
