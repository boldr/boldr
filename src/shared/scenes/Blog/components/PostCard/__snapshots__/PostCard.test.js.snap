// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> renders snapshot 1`] = `
<div
  className="boldrui-post-card"
>
  <Card
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardMedia
      overlay={
        <CardTitle
          subtitle="Invalid Date"
          title="Test Post"
        />
      }
    >
      <img
        alt="Test Post feature image"
      />
    </CardMedia>
    <CardText>
      <Row
        componentClass="div"
      >
        <Col
          componentClass="div"
          xs={12}
        >
          <Link
            className="readmore-link"
            replace={false}
            to="/blog/undefined"
          >
            <ButtonTooltipedInked
              inkTransitionEnterTimeout={450}
              inkTransitionLeaveTimeout={300}
              inkTransitionOverlap={150}
              label="Read More"
              onClick={[Function]}
              primary={true}
              raised={true}
            />
          </Link>
        </Col>
      </Row>
    </CardText>
    <CardActions>
      <TagBlock
        tags={null}
      />
    </CardActions>
  </Card>
</div>
`;
